x-pack/plugins/security/public/account_management/user_profile/user_profile_api_client.ts

```diff
@@ -51,8 +73,20 @@ export class UserProfileAPIClient implements UserProfileAPIClientType {
       .then((response) => {
         const data = response?.data ?? {};
         const updated = merge(this._userProfile$.getValue(), data);
+
         this._userProfile$.next(updated);
+        this._enabled$.next(true);
+        this._userProfileLoaded$.next(true);
+
         return response;
+      })
+      .catch((err) => {
+        // If we receive a 404 on the request, it means there are no user profile for the user.
+        const notFound = err?.response?.status === 404;
+        this._enabled$.next(notFound ? false : true);
+        this._userProfileLoaded$.next(true);
```

Comment:
I don't think there are any Error cases where we would want to display this toggle; if there wasn't a UserProfile loaded successfully, we shouldn't display the toggle. Is there a case you had in mind?

Currently if there was a 500 on this call, the toggle still shows up, but doesn't work as intended.

Reply:
I think it is good to be precise. An HTTP request can fail for multiple reason. We have [a specific error thrown by the server](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security/server/routes/user_profile/get_current.ts#L45) when there is no profile. We should limit it to that error status code.

If there is a 500 thrown, that does not mean that there is no user profile. Anything could happen and maybe on subsequent request it will not throw.